sdk/core/System.ClientModel/gen/PartialModifierAnalyzer.cs

```diff
@@ -0,0 +1,61 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System.Collections.Immutable;
+using System.Linq;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.Diagnostics;
+
+namespace System.ClientModel.SourceGeneration;
+
+/// <summary>
+/// Analyzer that checks if classes inheriting from ModelReaderWriterContext are declared with the partial modifier.
```

Comment:
You _could_ do this for a slightly nicer experience, but personally I prefer to just let the compiler give the user an error that they have multiple definitions for the type.